### ISSUE TITLE
ARM: threads' privilege stack alignment is not optimal

### ISSUE THREAD

**Issue by ioannisg** [zephyrproject-rtos/zephyr#16887]
The following code is auto-generated by `gen_priv_stacks.py` script (which is currently **only** used by the **ARM** architecture):

```

priv_stack_decl_temp = ("static u8_t __used"
                        " __aligned(PRIVILEGE_STACK_ALIGN)"
                        " priv_stack_%x[PRIVILEGE_STACK_ALIGN];\n")
```

Thus, we align a thread's privilege stack with its size. This, is however, not really needed, since the privilege stack does not need to be "sandboxed" into an MPU region, (something that is needed for the User stack), so we do not need to be accounting for the requirement for alignment-with-size - enforced in ARMv7-M builds.

In other words, when determining the alignment requirement of the priv stack, we only need to account for a possible MPU stack guard in the lowest part of the stack.

This is minor bug, as it might eat some SRAM when placing the priv_stacks_data.noint section.
